source/extensions/filters/common/rbac/config.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "envoy/config/filter/http/rbac/v2/rbac.pb.h"
+#include "envoy/config/filter/network/rbac/v2/rbac.pb.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "extensions/filters/common/rbac/engine_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace RBAC {
+
+/**
+ * All stats for the RBAC filter. @see stats_macros.h
+ */
+// clang-format off
+#define ALL_RBAC_FILTER_STATS(COUNTER)                                                             \
+  COUNTER(allowed)                                                                                 \
+  COUNTER(denied)                                                                                  \
+  COUNTER(shadow_allowed)                                                                          \
+  COUNTER(shadow_denied)
+// clang-format on
+
+/**
+ * Wrapper struct for RBAC filter stats. @see stats_macros.h
+ */
+struct RoleBasedAccessControlFilterStats {
+  ALL_RBAC_FILTER_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+enum class EnforcementMode { Enforced, Shadow };
+
+class RoleBasedAccessControlRouteSpecificFilterConfig : public Router::RouteSpecificFilterConfig {
```

Comment:
these two classes are pretty similar, you might want to combine them, or extract common part out.

Reply:
It's copied directly from the existing implementation. I remember there had been some discussions in previous PR to combine them but later decided to not to do so. Also looked at the content, it seems there is not too much thing to share, do you have any specific suggestions here?

Reply:
hmm, ok then I would not combine those config classes here, rather keep them separately.